### ISSUE TITLE
Error: Package “culori” could not be retrieved

### ISSUE THREAD

**Issue by danburzo** [Cykelero/tasklemon#1]
With the following script:

__myscript.js__
```js
cli.accept({ color: '-c' });
let { formatter, lab, rgb } = npm.culori;
formatter('hex')(lab(rgb(cli.args.color)));
```

Running it with `lemon myscript.js -c "#fff"` I get the error:

<details>
<summary>UnhandledPromiseRejectionWarning: Error: Package “culori” could not be retrieved: the package cache bundle preparation process is failing. Make sure the names of your packages are correct.</summary>

<pre>Preparing packages...
(node:20322) UnhandledPromiseRejectionWarning: Error: Package “culori” could not be retrieved: the package cache bundle preparation process is failing. Make sure the names of your packages are correct.
    at Object._getFromBundle (/usr/local/lib/node_modules/tasklemon/source/PackageCache.js:113:11)
    at Object.get (/usr/local/lib/node_modules/tasklemon/source/PackageCache.js:45:19)
    at Object.get (/usr/local/lib/node_modules/tasklemon/source/injected-modules/npm.js:6:23)
    at /private/var/folders/yy/h2wycn3j23n6lpf60qf3xmj40000gn/T/6jMhSY/rbg2lab.js:4:35
    at Object.<anonymous> (/private/var/folders/yy/h2wycn3j23n6lpf60qf3xmj40000gn/T/6jMhSY/rbg2lab.js:6:3)
    at Module._compile (internal/modules/cjs/loader.js:776:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:787:10)
    at Module.load (internal/modules/cjs/loader.js:643:32)
    at Function.Module._load (internal/modules/cjs/loader.js:556:12)
    at Module.require (internal/modules/cjs/loader.js:683:19)
(node:20322) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:20322) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.</pre>
</details>

I'm on MacOS Mojave with Node `v12.6.0`, npm `6.10.2` and yarn `1.17.3`. 
(The error reproduces with installing TL globally with either yarn or npm.)

**Comment by Cykelero** [comment]
Thanks for reporting this!
Right now, the install process in `preparePackageCacheBundle.js` isn't logging enough information for this to be debugged, so I'll start by making that better; it should allow us to understand what's going on.

**Comment by Cykelero** [comment]
OK—I think the error might be in creating the cache folder itself (my code foolishly assumes that `~/.cache` already exists, which probably is rarely the case on macOS).

Could you please try the fix I whipped up? Just checkout the [`improve-cache-folder-reliability`](https://github.com/Cykelero/tasklemon/tree/improve-cache-folder-reliability) branch, and run your script with that version of Tasklemon (getting it to run from source is trivial, as described in [Contributing](https://github.com/Cykelero/tasklemon#contributing)).

If this works, I'll make it into the next patch :)

**Comment by danburzo** [comment]
Hah, turns out `npx` can run off a GH branch, so this works:

```bash
npx Cykelero/tasklemon#improve-cache-folder-reliability myscript.js -c '#fff'
```

And I no longer get the error, so that must have been it. Cool! :-)

**Comment by Cykelero** [comment]
Perfect, thanks! This'll make it into a patch soon.

**Comment by Cykelero** [comment]
The fix has been released in [version 0.2.2](https://github.com/Cykelero/tasklemon/releases/tag/v0.2.2)!
